### PR TITLE
Remove useless code and refactor some Component methods

### PR DESCRIPTION
**Pull request by liufengyun**
When I'm writing the documentation, I found the core class `Component` can be simplified by removing useless fields and methods.

The following two methods are never used.

``` scala
final def send[T](name: String, t: T)
final def send[T](name: String, t: List[T])
```

### CODE REVIEW

**Comment by liufengyun** [comment]
I also make a small improvement by adding `start` and `stop` to the class `Component`, so that subclass doesn't have to handle the message `StartSignal` and `StopSignal`. Instead, they can just override `start` and `stop` if they want to do so.

**Comment by merlinND** [comment]
Cool refactoring, thanks for looking out for code quality!
Tell me when this PR is complete and we'll merge.

**Comment by liufengyun** [comment]
I'm planning to refactor the addDestination method a little so that the
users don't have to call it three times in order to register three message
types. It will make the API more friendly to our users.
Le 5 avr. 2015 09:49, "Merlin Nimier-David" notifications@github.com a
écrit :

**Comment by merlinND** [comment]
Sounds great!

**Comment by liufengyun** [comment]
I'm done with this pull request. I renamed `addDestination` to `->`, not sure if it's the thing that you would like.

File: ts/src/main/scala/ch/epfl/ts/component/Component.scala [hunk 1]
```diff
@@ -36,15 +36,19 @@ final class ComponentBuilder(name: String) {
...
     instances.head
   }
 }
 
-class ComponentRef(val ar: ActorRef, val clazz: Class[_], val name: String) {
+class ComponentRef(val ar: ActorRef, val clazz: Class[_], val name: String, cb: ComponentBuilder) {
   // TODO: Verify clazz <: Component
-  def addDestination(destination: ComponentRef, data: Class[_])(implicit cb: ComponentBuilder) = {
-    cb.add(this, destination, data)
+  def ->(destination: ComponentRef, types: Class[_]*) = {
```
Comment (merlinND) [inline ts/src/main/scala/ch/epfl/ts/component/Component.scala:52]: Very expressive syntax :+1: 

Could you add JavaDoc to explain this function's behavior?

Comment (liufengyun) [inline ts/src/main/scala/ch/epfl/ts/component/Component.scala:52]: OK, I've added Scala doc in a116c68050a42062429dee7186e52a98eee9472c.

**Comment by merlinND** [comment]
:shipit:

**Pull request merged**